### PR TITLE
Add table aws_guardduty_threat_intel_set.closes #243

### LINKED ISSUE #243: Add table aws_guardduty_threat_intel_set.
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/guardduty

### PR DESCRIPTION
**Pull request by Priyanka585464**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_guardduty_threat_intel_set []

PRETEST: tests/aws_guardduty_threat_intel_set

TEST: tests/aws_guardduty_threat_intel_set
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_guardduty_detector.primary: Creating...
aws_s3_bucket.bucket: Creating...
aws_guardduty_detector.primary: Creation complete after 3s [id=d8bc4134bcb43c2b62f390000d7ba85a]
aws_s3_bucket.bucket: Still creating... [10s elapsed]
aws_s3_bucket.bucket: Still creating... [20s elapsed]
aws_s3_bucket.bucket: Creation complete after 21s [id=terraform-20210330091410407000000001]
aws_s3_bucket_object.MyThreatIntelSet: Creating...
aws_s3_bucket_object.MyThreatIntelSet: Creation complete after 4s [id=MyThreatIntelSet]
aws_guardduty_threatintelset.named_test_resource: Creating...
aws_guardduty_threatintelset.named_test_resource: Still creating... [10s elapsed]
aws_guardduty_threatintelset.named_test_resource: Creation complete after 10s [id=d8bc4134bcb43c2b62f390000d7ba85a:00bc4134ebae6b2e7c47ed47709d0ddf]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
detector_id = d8bc4134bcb43c2b62f390000d7ba85a
region_name = us-east-1
resource_aka = arn:aws:guardduty:us-east-1:986325076436:detector/d8bc4134bcb43c2b62f390000d7ba85a/threatintelset/00bc4134ebae6b2e7c47ed47709d0ddf
resource_id = 00bc4134ebae6b2e7c47ed47709d0ddf
resource_name = turbottest60159

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "format": "TXT",
    "name": "turbottest60159",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest60159"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "account_id": "986325076436",
    "format": "TXT",
    "name": "turbottest60159",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest60159"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "format": "TXT",
    "name": "turbottest60159",
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest60159"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:guardduty:us-east-1:986325076436:detector/d8bc4134bcb43c2b62f390000d7ba85a/threatintelset/00bc4134ebae6b2e7c47ed47709d0ddf"
    ],
    "title": "turbottest60159"
  }
]
✔ PASSED

POSTTEST: tests/aws_guardduty_threat_intel_set

TEARDOWN: tests/aws_guardduty_threat_intel_set

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  detector_id,
  threat_intel_set_id,
  name,
  format,
  location
from
  aws_guardduty_threat_intel_set;
```
Screenshot 2021-03-31 at 1 54 38 PM

### List threat intel sets which are disabled

```sql
select
  threat_intel_set_id,
  status
from
  aws_guardduty_threat_intel_set
where
  status = 'INACTIVE';
```
```
+----------------------------------+--------+
| threat_intel_set_id              | status |
+----------------------------------+--------+
| 2cbc40ff8864d89ac1ea6b6d643481b0 | INACTIVE |
+----------------------------------+--------+
```

### CODE REVIEW

File: aws/plugin.go [hunk 1]
```diff
@@ -71,6 +71,8 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_eks_cluster":                        tableAwsEksCluster(ctx),
 			"aws_elasticache_cluster":                tableAwsElastiCacheCluster(ctx),
 			"aws_eventbridge_rule":                   tableAwsEventBridgeRule(ctx),
+			"aws_guardduty_detector":                 tableAwsGuardDutyDetector(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:74]:
```suggestion
```

File: aws/table_aws_guardduty_detector.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+				Description: "The last-updated timestamp for the detector.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getGuardDutyDetector,
+			},
+			{
+				Name:        "data_sources",
+				Description: "Describes which data sources are enabled for the detector.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getGuardDutyDetector,
+			},
```
Comment (Paulami30) [inline aws/table_aws_guardduty_detector.go:74]:
```suggestion
			},
			
```

File: aws/table_aws_guardduty_detector.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
+package aws
```
Comment (Paulami30) [inline aws/table_aws_guardduty_detector.go:1]: remove this file

File: aws/table_aws_guardduty_threat_intel_set.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+			},
+			{
+				Name:        "detector_id",
+				Description: "The ID of the detector.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getGuardDutyThreatIntelSet,
+				Transform:   transform.FromField("DetectorID"),
+			},
+			{
+				Name:        "name",
```
Comment (Paulami30) [inline aws/table_aws_guardduty_threat_intel_set.go:50]: Make this the first column

File: aws/table_aws_guardduty_threat_intel_set.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+				Description: "The URI of the file that contains the ThreatIntelSet.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getGuardDutyThreatIntelSet,
+			},
+			{
+				Name:        "status",
+				Description: "The status of threatIntelSet file uploaded.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getGuardDutyThreatIntelSet,
+			},
```
Comment (Paulami30) [inline aws/table_aws_guardduty_threat_intel_set.go:72]:
```suggestion
			},
			
```

**Comment by Priyanka585464** [comment]
This table will not work untill `Guardduty > detector` will merge into main branch.

File: aws/plugin.go [hunk 6]
```diff
@@ -81,6 +81,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_elasticache_replication_group":      tableAwsElastiCacheReplicationGroup(ctx),
 			"aws_emr_cluster":                        tableAwsEmrCluster(ctx),
 			"aws_eventbridge_rule":                   tableAwsEventBridgeRule(ctx),
+			"aws_guardduty_threat_intel_set":         tableAwsGuarddutyThreatIntelSet(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:84]: sort the list

Comment (Paulami30) [inline aws/plugin.go:84]: sort the list

File: aws/service.go [hunk 7]
```diff
@@ -29,6 +29,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/elbv2"
 	"github.com/aws/aws-sdk-go/service/emr"
 	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/aws/aws-sdk-go/service/guardduty"
```
Comment (Paulami30) [inline aws/service.go:32]: sort the list

File: aws/service.go [hunk 8]
```diff
@@ -436,6 +437,17 @@ func ELBService(ctx context.Context, d *plugin.QueryData, region string) (*elb.E
 	return svc, nil
 }
 
+// GuardDutyService returns the service connection for AWS ELB GuardDuty service
```
Comment (Paulami30) [inline aws/service.go:440]:
```suggestion
// GuardDutyService returns the service connection for AWS GuardDuty service
```

File: aws/service.go [hunk 9]
```diff
@@ -436,6 +437,17 @@ func ELBService(ctx context.Context, d *plugin.QueryData, region string) (*elb.E
...
+func GuardDutyService(ctx context.Context, d *plugin.QueryData, region string) (*guardduty.GuardDuty, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed GuardDutyService")
+	}
+
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("guardduty-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*guardduty.GuardDuty), nil
+
```
Comment (Paulami30) [inline aws/service.go:450]: where is the remaining part? And sort the function

File: aws/table_aws_guardduty_threat_intel_set.go [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+	}
+
+	return threatIntelSetInfo{*op, id, detectorID}, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func getAwsGuardDutyThreatIntelSetAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsGuardDutyThreatIntelSetAkas")
+	id := h.Item.(threatIntelSetInfo)
```
Comment (Paulami30) [inline aws/table_aws_guardduty_threat_intel_set.go:180]:
```suggestion
	data := h.Item.(threatIntelSetInfo)
```

File: aws/table_aws_guardduty_threat_intel_set.go [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+	// data := h.Item.(threatIntelSetInfo)
```
Comment (Paulami30) [inline aws/table_aws_guardduty_threat_intel_set.go:181]:
```suggestion
```

File: aws/table_aws_guardduty_threat_intel_set.go [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := c.(*awsCommonColumnData)
+	aka := "arn:" + commonColumnData.Partition + ":guardduty:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":detector" + "/" + id.DetectorID + "/threatintelset/" + id.ThreatIntelSetID
```
Comment (Paulami30) [inline aws/table_aws_guardduty_threat_intel_set.go:188]:
```suggestion
	aka := "arn:" + commonColumnData.Partition + ":guardduty:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":detector" + "/" + data.DetectorID + "/threatintelset/" + data.ThreatIntelSetID
```

File: docs/tables/aws_guardduty_threat_intel_set.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  threat_intel_set_id,
+  name,
+  format,
+  location
+from
+  aws_guardduty_threat_intel_set;
+```
+
+
+### List threatintelsets which are enabled
```
Comment (Paulami30) [inline docs/tables/aws_guardduty_threat_intel_set.md:21]:
```suggestion
### List threat intel sets which are disabled
```

Comment (Paulami30) [reply docs/tables/aws_guardduty_threat_intel_set.md:21]: modify the query accordingly

File: docs/tables/aws_guardduty_threat_intel_set.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+select
+  threat_intel_set_id,
+  status
+from
+  aws_guardduty_threat_intel_set
+where
+  status = 'ACTIVE';
+```
+
+
```
Comment (Paulami30) [inline docs/tables/aws_guardduty_threat_intel_set.md:33]:
```suggestion
```

**Review by Paulami30 — COMMENTED** [review]
Check for the comments please

File: aws/service.go [hunk 15]
```diff
@@ -503,6 +527,28 @@ func GlacierService(ctx context.Context, d *plugin.QueryData, region string) (*g
...
+		return nil, fmt.Errorf("region must be passed GuardDutyService")
+	}
+
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("guardduty-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*guardduty.GuardDuty), nil
+	}
+	
+	// so it was not in cache - create service
```
Comment (Paulami30) [inline aws/service.go:542]:
```suggestion
// so it was not in cache - create service
```

File: aws/plugin.go (hunk 6 above)
Comment (Paulami30) [reply aws/plugin.go:84]: still not sorted

**Review by Paulami30 — APPROVED** [review]

**Comment by Paulami30** [comment]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**